DOC: split usage into usage and developer-guide (#171)
audeering/audbackend: docs/usage.rst

    import audeer

    _cwd_root = os.getcwd()
    _tmp_root = audeer.mkdir('docs', 'tmp-usage')
    os.chdir(_tmp_root)

.. _usage:

Usage
=====

With the help of :mod:`audbackend`
a user can store files
in a repository

> on a storage system.

Reviewer: ```suggestion
on a storage system
(backend).
```

> The user can select between different :ref:`interfaces <interfaces>`
> that influence how the data is structured
> and presented to the user.

Reviewer: ```suggestion
File access is handled
via an :ref:`interface <interfaces>`,
which defines how the data is structured
and presented to the user.
```

In addition,
:mod:`audbackend` supports different storage systems,
so called :ref:`backends <backends>`.

> The names of all available backends can be listed by:

Reviewer: ```suggestion
The names of all available backends can be listed with:
```

Reviewer: At least sounds better to me, but I am not 100% sure :)

Author: I also had first "with", but it sounded to German to me ;)

I searched a littel bit and "listed by" seems to be used to refer to persons oding the listing, whereas "listed with" refers to objects that are listed together with the object before. So maybe "by" is then better as the code that follows is responsible for the listing and is not listed together with the text before?

Reviewer: Ok, let's stick with "by" then :)

.. jupyter-execute::

    import audbackend

    list(audbackend.available())

.. _unversioned-data-on-a-file-system:

Unversioned data on a file system
---------------------------------

> As all data is stored in a repository
> on the backend,
> we need to first create one.

Reviewer: ```suggestion
To store data on a backend
we need to create a repository first.
```

We select the ``'file-system'`` backend.

.. jupyter-execute::
    :hide-output:

    audbackend.create('file-system', './host', 'repo')

Once we have an existing repository,
we can access it with :func:`audbackend.access`.
We choose the :class:`audbackend.interface.Unversioned` interface
to communicate with the repository,
instead of the default one

> (:class:`audbackend.interface.Versioned`).

Reviewer: Is it important to mention the default here? Though it's probably obvious I think we should explain what unversioned access means, e.g. "We use the Unversioned interface. It does not support versioning, i.e. exactly one file exists for a backend path."

Author: I adjusted the text accordingly.

        display(str(ex.exception))

.. _versioned-data-on-a-file-system:

Versioned data on a file system
-------------------------------

We start by creating a repository

> on the ``'file-system'`` backend
> and accessing it
> using the default :class:`audbackend.interface.Versioned` interface.

Reviewer: ```suggestion
on the ``'file-system'`` backend.
This time we access it
with the :class:`audbackend.interface.Versioned` interface
(which is also used by default).
```

.. jupyter-execute::

    audbackend.create('file-system', './host', 'repo')

>     interface = audbackend.access('file-system', './host', 'repo')

Reviewer: ```suggestion
    interface = audbackend.access(
        'file-system',
        './host',
        'repo',
        interface=audbackend.interface.Versioned,
    )
```

I would be verbose here.

> We then upload a file and assign it ``'1.0.0'``
> as its version.

Reviewer: ```suggestion
We then upload a file and 
assign version ``'1.0.0'`` to it.
```

Author: I changed to:

```
We then upload a file
and assign version ``'1.0.0'`` to it.
```

.. jupyter-execute::

    with open('file.txt', 'w') as file:
        file.write('Content v1.0.0')
    interface.put_file('file.txt', '/file.txt', '1.0.0')

Now we change the file for version ``'2.0.0'``.

.. jupyter-execute::

    with open('file.txt', 'w') as file:
        file.write('Content v2.0.0')
    interface.put_file('file.txt', '/file.txt', '2.0.0')

If we inspect the content of the repository
it will return a list of tuples

> containing name and version.

Reviewer: ```suggestion
containing file name and version.
```
